contracts/factory/RegistryFactory.sol

```diff
@@ -22,6 +22,26 @@ import { IERC7484 } from "../interfaces/IERC7484.sol";
 import { INexusFactory } from "../interfaces/factory/INexusFactory.sol";
 import { MODULE_TYPE_VALIDATOR, MODULE_TYPE_EXECUTOR, MODULE_TYPE_FALLBACK, MODULE_TYPE_HOOK } from "../types/Constants.sol";
 
+library AddressSort {
+    function sort(address[] storage arr) internal {
+        uint n = arr.length;
+        for (uint i = 1; i < n; i++) {
+            address key = arr[i];
+            uint j = i - 1;
+            
+            // Sort addresses by comparing them directly
+            while (j >= 0 && arr[j] > key) {
+                arr[j + 1] = arr[j];
+                if (j == 0) {
+                    break;
+                }
+                j--;
+            }
+            arr[j + 1] = key;
+        }
+    }
+}
+
```

Comment:
if we'll use this let's move it on a separate contract

Reply:
using LibSort from solday now. 
should note that we need to write unit test for this..

Reply:
what kind of test ? I think if the tests are not breaking it should be fine !